### PR TITLE
[SPARK-21616][SPARKR][DOCS] update R migration guide and vignettes

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

update R migration guide and vignettes

## How was this patch tested?

manually

### CODE REVIEW

File: R/pkg/tests/fulltests/test_Windows.R [hunk 1]
```diff
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+
```
Comment (HyukjinKwon) [inline R/pkg/tests/fulltests/test_Windows.R]: We could remove this back tho :).

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by felixcheung** [comment]
Maybe. Other test files have that extra empty line though ;)

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**